app/controllers/well_known/oauth_metadata_controller.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module WellKnown
+  class OauthMetadataController < ActionController::Base # rubocop:disable Rails/ApplicationController
+    include CacheConcern
+
+    # Prevent `active_model_serializer`'s `ActionController::Serialization` from calling `current_user`
+    # and thus re-issuing session cookies
+    serialization_scope nil
+
+    def show
+      expires_in 3.days, public: true
+      render_with_cache json: ::OauthMetadataPresenter.new, serializer: ::OauthMetadataSerializer, content_type: 'application/json'
```

Comment:
Will this cache be invalidated when upgrading mastodon versions? If not, we may need to do something alternative here regarding caching.

Reply:
No, updating Mastodon generally does not invalid cache. Also there are two layers of caching here, the Rails cache, for 3 minutes, through `render_with_cache`, and the reverse-proxy cache through the headers set in `expires_in 3.days, public: true`

Reply:
Mmm, we may want to make this not have the `expires_in` such that it's always fresh, but bump `render_with_cache` to cache for longer.

Reply:
Have removed the `expires_in` and added the `expires_in:` option to `render_with_cache`, as to avoid issues during upgrades.